### PR TITLE
Fix set-project-version.sh

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

The script doesn't update the parent version for protobuf-shaded module

*Modifications*

use `versions:update-parent` to update parent version

### CODE REVIEW

File: src/set-project-version.sh [hunk 1]
```diff
@@ -33,6 +33,9 @@ pushd $(dirname "$0")/..
 mvn versions:set -DnewVersion=$NEW_VERSION
 mvn versions:set -DnewVersion=$NEW_VERSION -pl buildtools
 mvn versions:set -DnewVersion=$NEW_VERSION -pl pulsar-sql/presto-distribution
+# install the new version of root pom local, so `update-parent` can update the right parent version
+mvn -am -pl . clean install -DskipTests
+mvn versions:update-parent -DallowSnapshots=true -DparentVersion=$NEW_VERSION -f protobuf-shaded/pom.xml
 mvn versions:set -DnewVersion=$NEW_VERSION -f protobuf-shaded/pom.xml
```
Comment (jiazhai) [inline src/set-project-version.sh]: If update-parent worked well, then we could remove this line?  @merlimat

Comment (sijie) [reply src/set-project-version.sh]: no. we need both.

update-parent is to update the parent version; `set` is to set the version.

**Comment by jiazhai** [comment]
run integration tests

**Comment by jiazhai** [comment]
run integration tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
9160c14

**Pull request merged**